lib/filesystem.nix

```diff
@@ -42,4 +44,24 @@
               type = (builtins.readDir parent).${base} or null;
           in file == /. || type == "directory";
     in go (if isDir then file else parent);
+
+  # filterMapDir :  filter : ( name : String -> type : String -> Bool )
+  #              -> f : (name : String -> type : String -> a )
+  #              -> path : Path
+  #              -> { path : Path = a }
+  # This maps a function `f` over the entries filtered by `filter` in a directory located at `path`. 
+  # It doesn't recurse over subdirectories.
+  filterMapDir = filter: f: path:
+    #Sets are lexically sorted by key
+    let files = builtins.readDir path;
+        matches = lib.filterAttrs filter files;
+    in
+      builtins.mapAttrs (n: v: f (path + ("/" + n)) v) matches;
+
+  # mapDirFiles :  f : ( name : String -> type : String -> a )
+  #             -> { path : Path = a }
+  # This maps a function `f` over the "regular" files in a directory located at `path`.
+  # It doesn't recurse over subdirectories.
+  mapDirFiles = f: path:
+    mapDir (n: v: v == "regular") (n: v: f n) path;
```

Comment:
You renamed `mapDir`, so this won't work anymore

Reply:
D'oh.

Reply:
I fixed that and also the name in default.nix